internal/controller/metrics.go

```diff
@@ -0,0 +1,35 @@
+package controller
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+const (
+	metricsNamespace = "nyallocator"
+)
+
+var (
+	CurrentNodesVec = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: metricsNamespace,
+		Name:      "current",
+		Help:      "Number of nodes under the NodeTemplate",
+	}, []string{"name"})
+	ExpectedNodesVec = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: metricsNamespace,
+		Name:      "expected",
+		Help:      "Number of nodes specified in the NodeTemplate",
```

Comment:
I'm not confident, but in this case `desired` is suitable.
This value is a request and it is not as certain as `expected`.

```suggestion
	DesiredNodesVec = prometheus.NewGaugeVec(prometheus.GaugeOpts{
		Namespace: metricsNamespace,
		Name:      "desired_〇〇",
		Help:      "Number of nodes specified in the NodeTemplate",
```